### PR TITLE
Install Istio with istio-operator

### PR DESCRIPTION
**Pull request by Laci21**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

Istio was previously installed in an old-fashioned way, using Istio's Helm charts. Instead in this PR, we install [istio-operator](https://github.com/banzaicloud/istio-operator) first, then we create an Istio CR which triggers the istio-operator to install Istio.

### Additional context

- The istio-operator is created with `wait: True` so that the operator is ready by the time the Istio CR is created.
- The same three options can be configured for now as before: `mtls`, `autoSidecarInjectNamespaces` and `bypassEgressTraffic`.
- The API can be called the same way as before:
```json
  "postHooks": {
    "InstallServiceMesh": {
      "mtls": true,
      "autoSidecarInjectNamespaces": [
          "default"
      ],
      "bypassEgressTraffic": true
    }
  }
```
- When successfully created you should see something similar to this:
```bash
$ kubectl -n=istio-system get pod
NAME                                     READY     STATUS    RESTARTS   AGE
istio-citadel-7f485888b5-sz9dp           1/1       Running   0          41m
istio-egressgateway-b98b5495f-jhtfb      1/1       Running   0          41m
istio-galley-6676655-pkkwh               1/1       Running   0          41m
istio-ingressgateway-7d7b58558c-5thbn    1/1       Running   0          41m
istio-operator-operator-0                2/2       Running   0          59m
istio-pilot-5945c9dcc9-sjqpp             2/2       Running   0          41m
istio-policy-7dd44f8bd7-p2drq            2/2       Running   0          41m
istio-sidecar-injector-7787bcbd7-chmrd   1/1       Running   0          41m
istio-telemetry-74b77c89bd-gk5xh         2/2       Running   0          41m
```

### Checklist

- [x] Implementation tested (on GKE)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] OpenAPI and Postman files updated, client regenerated (`make generate-client`) (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

**Comment by tothandras** [comment]
Can't `InstallIstioOperator` accept itself the same options? It's a bit weird that you need both `InstallIstioOperator` and `InstallServiceMesh`. I would deprecate `InstallServiceMesh` eventually or just remove it.

**Comment by Laci21** [comment]
> Can't `InstallIstioOperator` accept itself the same options? It's a bit weird that you need both `InstallIstioOperator` and `InstallServiceMesh`. I would deprecate `InstallServiceMesh` eventually or just remove it.

Reduced to one posthook. The posthook can be called the same way as before.

File: cluster/istio_operator.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const istioOperatorNamespace = "istio-system"
+const istioOperatorDeploymentName = pkgHelm.BanzaiRepository + "/" + "istio-operator"
+const istioOperatorReleaseName = "istio-operator"
+
+// InstallIstioOperator installs istio-operator on a cluster
+func InstallIstioOperator(cluster CommonCluster) error {
```
Comment (sagikazarmark) [inline cluster/istio_operator.go]: This shouldn't be exported if not used as a posthook function.

File: cluster/istio_operator.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		true)
+	if err != nil {
+		return emperror.Wrap(err, "installing istio-operator with helm failed")
+	}
+
+	return nil
+}
+
+// CreateIstioCR creates an istio-operator specific CR which triggers the istio-operator to install Istio
+func CreateIstioCR(kubeConfig []byte, params *InstallServiceMeshParams, cluster CommonCluster) error {
```
Comment (sagikazarmark) [inline cluster/istio_operator.go]: This shouldn't be exported if not used as a posthook function.

**Pull request merged**